### PR TITLE
Add recent downloads

### LINKED ISSUE #702: Sort crates by number of downloads in the last 90 days
Part of [RFC 1824](https://github.com/rust-lang/rfcs/blob/master/text/1824-crates.io-default-ranking.md), rust-lang/rust#41616.

The decision made in RFC 1824 was to change the default ordering for crates within keywords and categories to be the number of downloads in the last 90 days rather than the number of downloads for all time.

An implementation question unresolved in that RFC: should we *change* the current sort option "Downloads" to always be "Downloads in the last 90 days" and remove the ability to sort by all-time downloads at all (Option 1), or should we *add* a new option to all sorting menus for downloads in the last 90 days and *keep* the option to sort by all-time downloads (Option 2)?

What needs to be done for either option:

For pages like [crates in the category "Data structures"](https://crates.io/categories/data-structures) or [crates with the keyword "ffi"](https://crates.io/keywords/ffi) (and notably NOT for a [search result page](https://crates.io/search?q=test) which should remain sorted by relevance by default, or the [Browse All Crates page](https://crates.io/crates), which should remain sorted alphabetically by default):

- Change the default ordering from all-time downloads to downloads in the last 90 days only. Actual implementation will depend on which option we go with, but [the diesel query being built here](https://github.com/rust-lang/crates.io/blob/102a91b61517bc0c1907b436fd3a70c9f27f517b/src/krate.rs#L670-L671) will need to be modified at some point.

## What needs to be done if we go with Option 1 (**replacing** sorting by all-time downloads with sorting by downloads in the last 90 days):

- Change the label of the sort, everywhere, from "Downloads" to "Recent Downloads" or similar (fix CSS if necessary to continue to look nice wider)
- Change the number of downloads displayed with each crate, everywhere, to be the number of downloads in the last 90 days so that the numbers match the sort
- Keep the `sort=downloads` query parameter, and just change what it does to the diesel query [here](https://github.com/rust-lang/crates.io/blob/102a91b61517bc0c1907b436fd3a70c9f27f517b/src/krate.rs#L670-L671) and [here](https://github.com/rust-lang/crates.io/blob/102a91b61517bc0c1907b436fd3a70c9f27f517b/src/krate.rs#L682-L683)

## What needs to be done if we go with Option 2 (**adding** sorting by downloads in the last 90 days to the sorting options and **keeping** sorting by all-time downloads as well): 

- Add the label "Recent Downloads" or similar to all the sorting options, everywhere (fix CSS if necessary to continue to look nice wider)
- Either *switching* the display of the number of downloads next to each crate to the number of all-time downloads depending on which sort option you've chosen, or alternatively always displaying both the number of downloads in the last 90 days and the number of all time downloads next to every crate, labeled as such.
- Create a new query parameter `sort=recent-downloads` or similar and add code to test for that value and handle it in diesel [here](https://github.com/rust-lang/crates.io/blob/102a91b61517bc0c1907b436fd3a70c9f27f517b/src/krate.rs#L670-L671) and [here](https://github.com/rust-lang/crates.io/blob/102a91b61517bc0c1907b436fd3a70c9f27f517b/src/krate.rs#L682-L683)

Also add tests no matter what we go with.

Please weigh in if you have opinions on which option we go with! 

If you'd like to work on this and there haven't been any comments, you get to pick which option we go with! Please ask me if you have any questions :)

### PR DESCRIPTION
**Pull request by natboehm**
This addresses issue #702, sorting crates by downloads in the past 90 days. The implementation follows Option 2 detailed in the issue, adding a new option to all sorting menus for downloads in the last 90 days and keeping the option to sort by all-time downloads. The number of recent downloads has been added to display below all-time downloads next to each crate, labeled accordingly.

This PR is currently waiting on updates to Diesel, without which it will not function correctly.
( edit: updates to Diesel complete :) )

Fixes #702

### CODE REVIEW

**Comment by vignesh-sankaran** [comment]
@natboehm: Could you run a `cargo fmt` against the codebase please? :)

**Comment by natboehm** [comment]
@vignesh-sankaran forgot about that one, will do!

**Comment by vignesh-sankaran** [comment]
@natboehm Thanks for that :). Which version of Diesel were you waiting on in order for this PR to work?

**Comment by natboehm** [comment]
@vignesh-sankaran not sure about the version but I think we're waiting on [this PR](https://github.com/diesel-rs/diesel/pull/1026) to be merged.

**Comment by sgrif** [comment]
I'm going to release tomorrow

**Comment by vignesh-sankaran** [comment]
@sgrif Were you also going to update the Diesel version that crates.io specifies?

**Comment by sgrif** [comment]
@natboehm If you rebase you should have everything you need now

File: src/krate.rs [hunk 1]
```diff
@@ -39,7 +40,17 @@ use util::{RequestUtils, CargoResult, internal, ChainError, human};
 use version::{EncodableVersion, NewVersion};
 use {Model, User, Keyword, Version, Category, Badge, Replica};
 
-#[derive(Debug, Clone, Queryable, Identifiable, AsChangeset)]
+#[derive(Insertable, Queryable, Identifiable, Associations, AsChangeset)]
```
Comment (sgrif) [inline src/krate.rs]: `#![deny(missing_debug_implementations, missing_copy_implementations)]` :)

**Pull request merged**